PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,367 @@
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[self.nextColor]
+            self.nextColor = (self.nextColor + 1) % len(drawable.palette)
+
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+
+        return cell_rect, cell_val
+
+    def display(self):
+        callEnviron = self.createCallEnvironment()
+        self.canvas.delete("all")
+
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+
+        # go through each Drawable in the list
+        for i, n in enumerate(self.list):
+            if i and n:
+                # create display objects for the associated Drawables
+                n.display_shape, n.display_val = self.createCellValue(
+                    i, n.val, n.color)
+                n.color = self.canvas.itemconfigure(n.display_shape, 'fill')
+
+        #self.index = self.createIndex(self.nItems-1, 'front', level=-1) # indicate priority
+        callEnviron |= set(self.index)
+
+        self.window.update()
+        
+    def newArraySize(self, val):
+        callEnviron = self.createCallEnvironment()                
+        # Clear Array and reset size and list
+        self.canvas.delete("all")
+        self.size = val
+        self.list = []        
+        
+        for i in range(val):  # Draw new grid of cells
+            self.createArrayCell(i) 
+        
+        self.window.update()
+        self.cleanUp(callEnviron)
+
+    # delete the last element of the queue, or None if empty
+    def remove(self):
+        callEnviron = self.createCallEnvironment()
+        if self.nItems == 0:
+            self.setMessage('Array is empty!')
+            return
+        self.startAnimations()
+        self.wait(0.3)
+
+        n = self.list[self.nItems - 1]
+
+        indexEnd = self.createIndex(len(self.list) - 1, 'front', level=-1)  # indicate priority
+        callEnviron |= set(indexEnd)
+
+        # Slide value rectangle up and off screen
+        items = (n.display_shape, n.display_val)
+        self.moveItemsOffCanvas(items, N, sleepTime=0.02)
+
+        self.nItems -= 1
+        self.moveItemsBy(indexEnd, (-self.CELL_SIZE, 0))  # move priority arrow
+
+        n = self.list.pop()
+        # delete the associated display objects
+        self.canvas.delete(n.display_shape)
+        self.canvas.delete(n.display_val)
+
+        self.cleanUp(callEnviron)
+
+    def fixCells(self):       # Move canvas display items to exact cell coords
+        for i, drawItem in enumerate(self.list):
```

Comment:
This was the main cause of the errors you mentioned when you tried to insert.  If you `enumerate(self.list)` you will get all elements in `self.list`.  That's initialized in the constructor to hold `self.size` cells filled with `None`.  This made `drawItem` be `None` on every operation (because `cleanUp()` is called which calls `fixCells()`.
Try replacing line 288 with:
```
        for i in range(self.nItems):
            drawItem = self.list[i]
```
This will iterate only over the first `nItems` of the `self.list`.  That's 0 initially and only grows when new items are inserted.